dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx

```diff
@@ -104,30 +108,43 @@ export function AppRepoForm(props: IAppRepoFormProps) {
         setFilterExclude(exclude);
         setFilterNames(names);
       }
-      if (secret) {
-        if (secret.data["ca.crt"]) {
-          setCustomCA(atob(secret.data["ca.crt"]));
+      if (repo.spec?.auth?.customCA || repo.spec?.auth?.header) {
+        const secrets = [];
+        if (repo.spec?.auth?.customCA) {
+          secrets.push(repo.spec.auth.customCA.secretKeyRef.name);
         }
-        if (secret.data.authorizationHeader) {
-          if (authHeader.startsWith("Basic")) {
-            const userPass = atob(authHeader.split(" ")[1]).split(":");
-            setUser(userPass[0]);
-            setPassword(userPass[1]);
-            setAuthMethod(AUTH_METHOD_BASIC);
-          } else if (authHeader.startsWith("Bearer")) {
-            setToken(authHeader.split(" ")[1]);
-            setAuthMethod(AUTH_METHOD_BEARER);
-          } else {
-            setAuthMethod(AUTH_METHOD_CUSTOM);
-            setAuthHeader(atob(secret.data.authorizationHeader));
-          }
+        if (repo.spec?.auth?.header && !secrets.includes(repo.spec.auth.header.secretKeyRef.name)) {
+          secrets.push(repo.spec.auth.header.secretKeyRef.name);
         }
-        if (secret.data[".dockerconfigjson"]) {
-          setAuthMethod(AUTH_METHOD_REGISTRY_SECRET);
+        secrets.forEach(s => dispatch(actions.repos.fetchRepoSecret(namespace, s)));
```

Comment:
Mmm, interesting, so in the AppRepoForm, once loaded, we are fetching:

1. The whole list of `imagePullSecrets` un a given namespace
2. The `repoSecret` each time a repo is used/selected/etc

no?

Reply:
On another tangent, this is a good example where an API for fetching secrets *without* the actual secret data being sent over the wire would be useful, IMO, and only fetching the actual secret data when required for a particular secret. But that's a goal for the new API design, not this PR :)